src/Provider/DefaultSmsProvider.php

```diff
@@ -52,15 +55,54 @@ public function __construct(ConfigFactoryInterface $config_factory, ModuleHandle
   /**
    * {@inheritdoc}
    */
-  public function send(SmsMessageInterface $sms, array $options = array()) {
-    // Check if a preferred gateway is specified in the $options.
-    if (isset($options['gateway'])) {
-      $gateway = SmsGateway::load($options['gateway']);
+  public function queue(SmsMessageEntityInterface $sms_message) {
+    $gateway = $this->getGateway($sms_message);
+    if ($gateway->getSkipQueue()) {
+      $this->send($sms_message, []);
+      return;
+    }
+
+    if (!$sms_message->getGateway()) {
+      // @fixme getGateway being falsy is undocumented...
+      $sms_message->setGateway($this->getDefaultGateway());
+    }
+
+    if ($count = $sms_message->validate()->count()) {
+      throw new SmsException(sprintf('Can not queue SMS message because there are %s validation error(s).', $count));
+    }
+
+    // Split messages to overcome gateway limits.
+    $max = $gateway->getMaxRecipientsOutgoing();
+    $recipients_all = $sms_message->getRecipients();
+    if ($max > 0 && count($recipients_all) > $max) {
+      foreach ($sms_message->chunkByRecipients($max) as $sms_message) {
+        $sms_message->save();
+      }
     }
```

Comment:
I think the chunking should happen in `::send()` since there are no restrictions to how many recipients you can save to storage.